### DISCUSSION TITLE
Is it Possible to Convert Email Input to Lowercase Before Querying the Database Without Customizing the Authentication Flow?

### DISCUSSION THREAD

**Discussion by ramonmalcolm10** [filamentphp/filament#9016] [Help] [answered]
I store all emails in lowercase in my Postgres database. However, I've noticed that if a user doesn't provide their email in lowercase during authentication, it fails. Is there a way to automatically convert the email to lowercase without having to customize the login flow?

**Comment by leandrocfe** [comment, marked as answer]
It is case-insensitive on mysql...
You could create a custom class in the app

```php
use Filament\Forms\Components\Component;
use Filament\Pages\Auth\Login as BaseLogin;

class Login extends BaseLogin
{
    protected function getEmailFormComponent(): Component
    {
        return parent::getEmailFormComponent()
            ->dehydrateStateUsing(fn ($state): string => strtolower($state));
    }
}

```
and use in the panel

```php
$panel
    ->login(Login::class)
    ...
```

**Reply by ramonmalcolm10** [reply]
Working. Thanks for the quick response.
